### PR TITLE
[gql] Report runless asset events

### PR DESCRIPTION
**Pull request by clairelin135**
This PR adds graphQL to enable reporting asset materializations in the UI. 

It creates a `reportRunlessAssetEvents` mutation with the following arguments:
- event type -- currently only accepts observations and materializations, though we can use this to support asset checks in the UI in the future
- asset key -- our planned UIs currently only support targeting one asset at a time
- partition keys (optional)
- description (optional)

In the next PR I plan on enabling displaying the creator of a runless event.

### CODE REVIEW

File: python_modules/dagster-graphql/dagster_graphql/schema/inputs.py [hunk 1]
```diff
@@ -188,6 +190,34 @@ class Meta:
...
+class GrapheneRunlessAssetEventType(graphene.Enum):
+    """The event type of an asset event."""
+
+    ASSET_MATERIALIZATION = "ASSET_MATERIALIZATION"
+    ASSET_OBSERVATION = "ASSET_OBSERVATION"
+
+    class Meta:
+        name = "AssetEventType"
+
+    def to_dagster_event_type(self):
```
Comment (sryza) [inline python_modules/dagster-graphql/dagster_graphql/schema/inputs.py:202]: Return type plz

**Review by sryza — APPROVED** [review]
Nice

**Comment by clairelin135** [comment]
Corresponding internal PR: https://github.com/dagster-io/internal/pull/7123

**Comment by clairelin135** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17019** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17019" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/17019?utm_source=stack-comment).

**Pull request merged**